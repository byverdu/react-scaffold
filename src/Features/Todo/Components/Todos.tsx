import React from 'react';
import { Map } from 'immutable';
import { Todo } from 'Models/Todo';
import TodoItem from 'Features/Todo/Components/TodoItem';

interface TodosProps {
  todos: Map<string, Todo>;
  checkHandler: (id: string, isChecked: boolean) => void;
  deleteTodo: (id: string) => void;
}

const Todos: React.SFC<TodosProps> = ({ todos, checkHandler, deleteTodo }) => {
  return (
    <ul>
      {todos.valueSeq().map((item) => (
        <TodoItem
          deleteTodo={deleteTodo}
          checkHandler={checkHandler}
          key={item.id}
          todo={item}
        />
      ))}
    </ul>
  );
};

export default Todos;
