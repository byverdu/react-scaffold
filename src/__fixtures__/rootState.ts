import { TodoState, VisibilityFilterState } from 'Models/Todo';
import { LoggerState } from 'Models/Logger';
import { RootState } from 'Core/reducers';
import { VisibilityFilterEnum } from 'Models/Enums';
import { Map } from 'immutable';
import { mockedData } from '__fixtures__/actions';

const todos: TodoState = {
  todos: Map(mockedData)
};

const visibilityFilter: VisibilityFilterState = {
  activeFilter: VisibilityFilterEnum.SHOW_ALL
};

export const logger: LoggerState = {
  type: undefined,
  message: '',
  isVisible: false
};

const rootState: RootState = {
  todos,
  visibilityFilter,
  logger
};

export default rootState;
