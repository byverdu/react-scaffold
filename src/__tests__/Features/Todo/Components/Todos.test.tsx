import React from 'react';
import rootState from '__fixtures__/rootState';
import Todos from 'Features/Todo/Components/Todos';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<Todos />', () => {
  const todos = rootState.todos.todos;
  const wrapper = mount(
    <Todos todos={todos} checkHandler={jest.fn()} deleteTodo={jest.fn()} />
  );

  it('should create a snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render as many TodoItem as the size of the state', () => {
    expect(wrapper.find('TodoItem')).toHaveLength(todos.size);
  });
});
