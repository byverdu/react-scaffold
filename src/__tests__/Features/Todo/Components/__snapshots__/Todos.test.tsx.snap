// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todos /> should create a snapshot 1`] = `
<Todos
  checkHandler={[MockFunction]}
  deleteTodo={[MockFunction]}
  todos={
    Immutable.Map {
      "6789": Object {
        "done": false,
        "id": 6789,
        "text": "invasion! ðŸ‘¾",
      },
      "12345": Object {
        "done": false,
        "id": 12345,
        "text": "oh yeah! ðŸ‘“",
      },
    }
  }
>
  <ul>
    <TodoItem
      checkHandler={[MockFunction]}
      deleteTodo={[MockFunction]}
      key="6789"
      todo={
        Object {
          "done": false,
          "id": 6789,
          "text": "invasion! ðŸ‘¾",
        }
      }
    >
      <li
        className="task-todo"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <label
          data-id={6789}
          htmlFor="list-item-6789"
        >
          <input
            checked={false}
            id="list-item-6789"
            onChange={[Function]}
            type="checkbox"
          />
          invasion! ðŸ‘¾
        </label>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
    </TodoItem>
    <TodoItem
      checkHandler={[MockFunction]}
      deleteTodo={[MockFunction]}
      key="12345"
      todo={
        Object {
          "done": false,
          "id": 12345,
          "text": "oh yeah! ðŸ‘“",
        }
      }
    >
      <li
        className="task-todo"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <label
          data-id={12345}
          htmlFor="list-item-12345"
        >
          <input
            checked={false}
            id="list-item-12345"
            onChange={[Function]}
            type="checkbox"
          />
          oh yeah! ðŸ‘“
        </label>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
    </TodoItem>
  </ul>
</Todos>
`;
